Research/README.md

```diff
@@ -1,6 +1,98 @@
+4. `RSDFormUIStep` defines one or more input fields.
+
+Additionally, both `RSDTask` and `RSDUIStep` inherit from `RSDUIActionHandler` which can be used to define the text, image, and action that would map to a given button. This allows customization of the UI/UX for a given button to be defined either for that specific step or globally for the entire task.
+
+### `RSDStepNavigator`
+
+The step navigator vends the steps and progress for a task. This framework includes one implementation of the step navigator which is the `RSDConditionalStepNavigator`. The conditional step navigator uses protocol extension to take an ordered list of steps, and check those steps against the input `RSDTaskResult` to determine whether or not a given step should be skipped or if navigation should jump to a different step using the step `identifier`.
+
+For the conditional step navigator, navigation is handled by querying both a list of `RSDTrackingRule` objects defined on `RSDFactory.shared` and by checking the steps for conformance to a set of navigation rules. Specifically, this framework includes:
+
+1. `RSDNavigationRule` which is can return a next step identifier that is then used to jump within the navigation.
+2. `RSDNavigationSkipRule` which allows the step to be skipped based on the current task result.
+3. `RSDNavigationBackRule` is used to block backward navigation.
+4. `RSDSurveyNavigationStep` defines navigation rules that this step would apply for a given task result based on the answer to input field entry.
+5. `RSDCohortNavigationStep` and `RSDCohortAssignmentStep` are used to navigate based on cohorts to which the participant has been added.
+
+The step results of navigating the task are added to the `RSDTaskResult` on the `stepHistory`. Results added to the step history are intended to be added sequentially in the order the steps were displayed to the user. A step result can be any result that conforms to the `RSDResult` protocol. These results should be uniquely defined by their `identifier` which matches the `identifier` for the step.
+
+### `RSDAsyncActionConfiguration`
+
+`RSDAsyncActionConfiguration` objects define any configuration required for running a background action. For example, recording motion sensor information or pinging a weather service. Most of the work of setting up and running async actions is left to the specific application.
+
+### `RSDResult`
+
+The results of running a task are described using `RSDResult` objects. The base protocol requires conformance to the `Encodable` protocol but does *not* require conformance to `Decodable`. This allows using class objects that cannot be extended to conform to the `Decodable` protocol, such as `ORKResult` classes.
+
+There are several protocols that inherit from `RSDResult` that are included in the core framework. These include:
+
+1. `RSDAnswerResult` which is used to describe an answer to an input field question.
```

Comment:
It seems like answer results are more useful than just for input field questions. For example the FirstRun boolean answer result in mPower doesn't come from user input. I think it would be good to mention that you can use them to stick arbitrary data into the task result as well.

Reply:
True, but in the core framework, that's how they are used. I was trying to keep it simple. ;)